lua/telescope/builtin/lsp.lua

```diff
@@ -120,20 +121,56 @@ lsp.code_actions = function(opts)
     return
   end
 
-  local _, response = next(results_lsp)
-  if not response then
-    print("No code actions available")
-    return
+  local idx = 1
+  local results = {}
+  local widths = {
+    idx = 0,
+    command_title = 0,
+    client_name = 0,
+  }
+
+  for client_id, response in pairs(results_lsp) do
+    if response.result then
+      local client = vim.lsp.get_client_by_id(client_id)
+
+      for _, result in pairs(response.result) do
+        local entry = {
+          idx = idx,
+          command_title = result.title,
+          client_name = client and client.name or "",
+          command = result,
+        }
+
+        for key, value in pairs(widths) do
+          widths[key] = math.max(value, vim.fn.strdisplaywidth(entry[key]))
+        end
+
+        table.insert(results, entry)
+        idx = idx + 1
+      end
+    end
   end
 
-  local results = response.result
-  if not results or #results == 0 then
+  if #results == 0 then
     print("No code actions available")
     return
   end
 
-  for i,x in ipairs(results) do
-    x.idx = i
+  local displayer = entry_display.create {
+    separator = " ",
+    items = {
+      { width = widths.idx },
+      { width = widths.command_title },
+      { width = widths.client_name },
+    },
+  }
+
+  local function make_display(entry)
+    return displayer {
+      {entry.idx, "TelescopePromptPrefix"},
```

Comment:
If we still want `1: ` than we should do `entry.idx .. ':'` here and `widths.idx + 1`.